fhirflat/ingest.py

```diff
@@ -0,0 +1,279 @@
+"""
+Stores the main functions for converting clinical data (initally from RedCap-ARCH) to
+FHIRflat.
+
+TODO: Eventually, this should link to a google sheet file that contains the mappings
+"""
+
+import pandas as pd
+import numpy as np
+import warnings
+import os
+from math import isnan
+from fhirflat.util import get_local_resource
+
+# 1:1 (single row, single resource) mapping: Patient, Encounter
+# 1:M (single row, multiple resources) mapping: Observation, Condition, Procedure, ...
+
+"""
+TODO
+* sort out reference formatting + how to choose ID's e.g. location within encounter etc
+* cope with 'if' statements - e.g. for date overwriting.
+* deal with duplicates/how to add multiple values to a single field - list options.
+* Consider using pandarallel (https://pypi.org/project/pandarallel/) to parallelize
+    the apply function, particularly for one to many mappings.
+"""
+
+
+def find_field_value(row, response, mapp, raw_data=None):
+    """
+    Returns the data for a given field, given the mapping.
+    For one to many resources the raw data is provided to allow for searching for other
+    fields than in the melted data.
+    """
+    if mapp == "<FIELD>":
+        return response
+    elif "+" in mapp:
+        mapp = mapp.split("+")
+        results = [find_field_value(row, response, m) for m in mapp]
+        results = [x for x in results if x == x]
+        return " ".join(results)
+    elif "if not" in mapp:
+        mapp = mapp.replace(" ", "").split("ifnot")
```

Comment:
should be 'if not'

Reply:
The spaces are stripped from the string first to remove the prefix & suffix spaces as well, hence the "ifnot" rather than "if not"